internal/charm/testing/charmdir_test.go

```diff
@@ -27,31 +28,53 @@ type CharmDirSuite struct {
 
 var _ = gc.Suite(&CharmDirSuite{})
 
-func (s *CharmDirSuite) TestIsCharmDirGoodCharm(c *gc.C) {
-	path := charmDirPath(c, "dummy")
-	c.Assert(charm.IsCharmDir(path), jc.IsTrue)
-}
-
-func (s *CharmDirSuite) TestIsCharmDirBundle(c *gc.C) {
-	path := bundleDirPath(c, "wordpress-simple")
-	c.Assert(charm.IsCharmDir(path), jc.IsFalse)
-}
-
-func (s *CharmDirSuite) TestIsCharmDirNoMetadataYaml(c *gc.C) {
-	path := charmDirPath(c, "bad")
-	c.Assert(charm.IsCharmDir(path), jc.IsFalse)
+func checkDummy(c *gc.C, f charm.Charm) {
```

Comment:
**suggestion:** Make the test fails,  look at the error and decide if this kind of pattern is convenient. I think it is not, because you don't get easily redirected to the actual broken test.

We should look more into gocheck to have the Assert consistently called directly from the test, with this degree of factorization.